File: src/Web/JavaScriptJS/patterns/repository.js
const assert = require("node:assert/strict");
/** Presents collection-like domain access while hiding persistence details. */
function run() {
  const rows = new Map();
  const users = {
    add(user) { rows.set(user.id, { ...user }); },
    get(id) { const row = rows.get(id); return row && { ...row }; },
  };
  users.add({ id: 1, name: "Ada" });
  assert.deepEqual(users.get(1), { id: 1, name: "Ada" });
  return "repository";
}
module.exports = { run };
if (require.main === module) run();

File: src/Scripting/Lua/patterns/repository.lua
local rows = { {id=1,name="one"}, {id=2,name="two"} }
local repository = {}
function repository:get(id)
  for _, row in ipairs(rows) do if row.id == id then return { id=row.id, name=row.name } end end
end
assert(repository:get(2).name == "two")
return true
